### PR TITLE
move test_constitute to RSpec

### PR DESCRIPTION
**Pull request by mockdeep**

### CODE REVIEW

File: spec/rmagick/image/constitute_spec.rb [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
+
+  it 'raises an error when invalid RGBA pixel values are given' do
+    pixels = ['x'] * (4 * img.columns * img.rows)
+    expected_message = 'element 0 in pixel array is String, must be numeric'
+
+    expect { Magick::Image.constitute(img.columns, img.rows, 'RGBA', pixels) }
+      .to raise_error(TypeError, expected_message)
+  end
+
+  it 'raises an error when 0 is passed for columns' do
```
Comment (dlemstra) [inline spec/rmagick/image/constitute_spec.rb]: Maybe also check for a negative value?

File: spec/rmagick/image/constitute_spec.rb [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+  end
+
+  it 'raises an error when invalid RGBA pixel values are given' do
+    pixels = ['x'] * (4 * img.columns * img.rows)
+    expected_message = 'element 0 in pixel array is String, must be numeric'
+
+    expect { Magick::Image.constitute(img.columns, img.rows, 'RGBA', pixels) }
+      .to raise_error(TypeError, expected_message)
+  end
+
```
Comment (dlemstra) [inline spec/rmagick/image/constitute_spec.rb]: Maybe we should also have a test for `wrong number of array elements`?

File: spec/rmagick/image/constitute_spec.rb (hunk 1 above)
Comment (mockdeep) [reply spec/rmagick/image/constitute_spec.rb]: @dlemstra I'm trying to add this, but we don't actually check for negatives in our code. Instead it complains in a later check:

```
ArgumentError: wrong number of array elements (60960 for -1440)
```

The check in the code only fails when either column is 0. I guess I can update the code, but we might want to do that in a separate PR.

File: spec/rmagick/image/constitute_spec.rb (hunk 2 above)
Comment (mockdeep) [reply spec/rmagick/image/constitute_spec.rb]: I added a test for wrong number of array elements.

**Pull request merged**